src/runtime/HalideRuntime.h

```diff
@@ -1364,10 +1364,10 @@ extern halide_can_use_target_features_t halide_set_custom_can_use_target_feature
 extern int halide_default_can_use_target_features(int count, const uint64_t *features);
 
 typedef struct halide_dimension_t {
-    int32_t min, extent, stride;
+    int32_t min, extent, stride;  // NOLINT(modernize-use-default-member-init)
```

Comment:
Since this file has to be usable from C as well as C++, we have to do something here. NOLINT is ugly but works. The alternative would be to make the entire interior of the class #ifdef __cplusplus. Better? Worse?

Reply:
Having a default value for the dimension fields seems like a plus when using this struct from C++ code, so I'd vote for the ifdef